src/entities/trades/curve/utils.ts

```diff
@@ -55,22 +71,32 @@ export function getCurveToken(tokenAddress: string, chainId: ChainId = ChainId.M
  * @param tokenOutAddress Token out address
  * @returns List of potential pools at which the trade can be done
  */
-export function getRoutablePools(pools: CurvePool[], tokenIn: CurveToken, tokenOut: CurveToken, chainId: ChainId) {
-  return pools.filter(({ tokens, metaTokens, underlyingTokens, allowsTradingETH }) => {
+export async function getRoutablePools(
+  pools: CurvePool[],
+  tokenIn: CurveToken,
+  tokenOut: CurveToken,
+  chainId: ChainId
+) {
+  const factoryPools = await Fetcher.fetchCurveFactoryPools(chainId)
```

Comment:
Do you want to add a simple in-memory cache for this? The quote estimate is fetched from the contract directly. However, pool information such as tokens in the pool will unlikely to change :)

Reply:
Yeah good suggestion will look into it! Only new pools added can change the response current ones will stay the same